src/test/java/com/synopsys/integration/detect/battery/docker/integration/BlackDuckAssertions.java

```diff
@@ -127,6 +130,26 @@ public void doesNotHaveComponents(Set<String> componentNames) throws Integration
         });
     }
 
+    public void checkComponentVersionExists(String componentName, String version) throws IntegrationException {
+        List<ProjectVersionComponentVersionView> bomComponents = projectBomService.getComponentsForProjectVersion(retrieveProjectVersionWrapper().getProjectVersionView());
+        Optional<ProjectVersionComponentVersionView> blackDuckCommonComponent = bomComponents.stream()
+                .filter(ProjectVersionComponentView -> componentName.equals(ProjectVersionComponentView.getComponentName()))
+                .findFirst();
+        assertTrue(blackDuckCommonComponent.isPresent());
+        String blackDuckCommonComponentVersion = blackDuckCommonComponent.get().getComponentVersionName();
+        assertEquals(version, blackDuckCommonComponentVersion);
+    }
+
+    public void checkComponentVersionNotExists(String componentName, String version) throws IntegrationException {
+        List<ProjectVersionComponentVersionView> bomComponents = projectBomService.getComponentsForProjectVersion(retrieveProjectVersionWrapper().getProjectVersionView());
```

Comment:
Looks like a lot of methods in this class rely on `bomComponents` from the same project/version.
I wonder if it's worth it to memoize these values to avoid redundant requests.